<a href="https://colab.research.google.com/github/Mechitex705/image_recognition/blob/main/image_recognition.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.image as mpimg
import tensorflow as ts
import os

import tensorflow as tf

# Define dataset directory
dataset_dir = '/content/drive/MyDrive/image_recognition_tb'

# Define image size and batch size
image_size = (512, 512)  # As your images are 512x512
batch_size = 32

# Load dataset, this will automatically label based on folder names
train_ds = tf.keras.preprocessing.image_dataset_from_directory(
    dataset_dir,
    image_size=image_size,
    batch_size=batch_size,
    label_mode='int',  # Automatically assigns labels based on folder names
    class_names=['Normal', 'Tuberculosis']  # Ensures order of labels is consistent
)

Found 4271 files belonging to 2 classes.


In [7]:
# Check the class names
print(train_ds.class_names)

# Inspect the shape and labels of a batch
for images, labels in train_ds.take(1):
    print(f'Image batch shape: {images.shape}')
    print(f'Label batch shape: {labels.shape}')


['Normal', 'Tuberculosis']
Image batch shape: (32, 512, 512, 3)
Label batch shape: (32,)


**4271** images have been successfully divided into 2 classes, Normal and Tuberculosis.
I loaded 32 images per batch, and each image is 512x512 pixels with 3 color channels.
Each batch comes with 32 corresponding labels (one for each image).